src/main/java/org/openstreetmap/atlas/geography/Polygon.java

```diff
@@ -378,31 +378,51 @@ public boolean isApproximatelyNSided(final int expectedNumberOfSides, final Angl
+     *      "http://www.gutenberg.org/files/19770/19770-pdf.pdf?session_id=374cbf5aca81b1a742aac0879dbea5eb35f914ea"></a>
+     * @see <a href="http://mathforum.org/library/drmath/view/51879.html"></a>
+     * @see <a href="http://mathforum.org/library/drmath/view/65316.html"></a>
      */
     public boolean isClockwise()
     {
-        double sum = 0;
-        EarthCenteredEarthFixedCoordinate previousCartesianPoint = null;
+        // Formula to calculate the area of triangle on a sphere is (A + B + C - Pi) * radius *
+        // radius.
+        // Equation (A + B + C - Pi) is called the spherical excess. We are going to divide our
+        // polygon in triangles and then calculate the signed area of each triangle. Sum of the
+        // areas of these triangles will be the area of this polygon
+        double sphericalExcess = 0;
+        Location previousLocation = null;
 
         for (final Location point : this.closedLoop())
         {
-            // This will return earth centered cartesian coordinates for a location
-            final EarthCenteredEarthFixedCoordinate newCartesianPoint = new EarthCenteredEarthFixedCoordinate(
-                    point);
+            final Location currentLocation = point;
 
-            if (previousCartesianPoint != null)
+            if (previousLocation != null)
             {
-                sum += (newCartesianPoint.getX() - previousCartesianPoint.getX())
-                        * (newCartesianPoint.getY() + previousCartesianPoint.getY());
+                // for the sake of simplicity we are using two vertices from the polygon and the
+                // third vertex would be North Pole.
+                // Please refer "Spherical Trigonometry by I.Todhunter".
+                // Section starting on page 7 and 17 for triangle identities and trigonometric
+                // functions.
+                // Also look on page 71 for getting the area of triangle
+                final double latitudeOne = previousLocation.getLatitude().asRadians();
+                final double latitudeTwo = currentLocation.getLatitude().asRadians();
+                final double deltaLongitude = currentLocation.getLongitude().asRadians()
+                        - previousLocation.getLongitude().asRadians();
+
+                final double alpha = Math
+                        .sqrt((1 - Math.sin(latitudeOne)) / (1 + Math.sin(latitudeOne)))
+                        * Math.sqrt((1 - Math.sin(latitudeTwo)) / (1 + Math.sin(latitudeTwo)));
+
+                // You can derive this from the formula on Page 74, point 102 of the book
+                sphericalExcess += 2 * Math.atan2(alpha * Math.sin(deltaLongitude),
+                        1 + alpha * Math.cos(deltaLongitude));
             }
-            previousCartesianPoint = newCartesianPoint;
+            previousLocation = currentLocation;
         }
-        return sum >= 0;
+
+        return sphericalExcess * Distance.AVERAGE_EARTH_RADIUS.asKilometers()
```

Comment:
Is multiplying by the earth radius twice here necessary? The result will always have the same sign as `sphericalExcess` anyway.

Reply:
Yes, I agree that it is not necessary but I kept it so that anyone reading this should know that we are calculating the area of the polygon

Reply:
Yes, I think you should remove it as it does not bring any more information. To let people know that we are calculating the area of a polygon, a comment should do it!

Reply:
okay, sounds good 👍